src/wallet/db.cpp

```diff
@@ -342,8 +348,61 @@ bool BerkeleyBatch::Recover(const fs::path& file_path, void *callbackDataIn, boo
         return false;
     }
 
-    std::vector<BerkeleyEnvironment::KeyValPair> salvagedData;
-    bool fSuccess = env->Salvage(newFilename, salvagedData);
+    /**
+     * Salvage data from a file. The DB_AGGRESSIVE flag is being used (see berkeley DB->verify() method documentation).
+     * key/value pairs are appendedto salvagedData which are then written out to a new wallet file.
+     * NOTE: reads the entire database into memory, so cannot be used
+     * for huge databases.
+     */
+    std::vector<KeyValPair> salvagedData;
+    bool fSuccess = false;
```

Comment:
in commit 6c3afd67d2:

Why is this set all the way up here? The outcome is only needed to remember whether the end of the stream was reached or not. I think it should be defined after the last write to `keyHex`, but still before this code block:

Also, you could leave it uninitialized because every if branch will initialize it. If none does the initialization, at least a sanitizer could yell at us and hint at the bug.
```cpp
    bool fSuccess;
    if (keyHex != DATA_END) {
        LogPrintf("Salvage: WARNING: Unexpected end of file while reading salvage output.\n");
        fSuccess = false;
    } else {
        fSuccess = (result == 0);
    }

Reply:
It was to reduce the amount of changes being done in a move commit.

Reply:
Changed.